### PR TITLE
Abstract file access layer

### LINKED ISSUE #111: [TBD] [async] Abstract storage API to access storage
⚠️ **To be discussed, don't work on this for now.**  
The main issue is that tokio async File API is slower writing to local files than std File API, even with `io_uring`. We will see if that will be resolved in the future, then we can think of doing this.

This needs to be like #247, just async.

Flow is like this
```
lib-users > crypto::fs_api::[fs, fs::OpenOptions, fs::File, io] > EncryptedFs -> crypto::storage::a_sync::[fs, fs::OpenOptions, fs::File, io]
```

This issue corresponds to `encrypted_storage::a_sync::[fs, fs::File, io]`.

Screenshot from 2024-09-22 18-06-49

### PR DESCRIPTION
**Pull request by someotherself**
Draft for issue 111.
Will add more docs and tests before merge.

### CODE REVIEW

File: src/crypto/cryptofs.rs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+/// // Create a file name
+/// let file_name = SecretString::from_str("file1").unwrap();
+/// // Call CryptoOptions
+/// let file = CryptoOptions::build(&fs).open(&file_name);
+/// // Use read operations on opened file
+/// }
+/// ```
+#[allow(dead_code)]
+pub struct CryptoOptions<'a, State> {
+    state: State,
```
Comment (radumarias) [inline src/crypto/cryptofs.rs]: call these as in the Rust API, like OpenOptions

File: src/crypto/cryptofs.rs [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    fs: &'a EncryptedFs,
+}
+
+#[derive(Debug)]
+pub struct File {
+    pub fh: u64,
```
Comment (radumarias) [inline src/crypto/cryptofs.rs]: like this is

File: src/crypto/cryptofs/test.rs [hunk 3]
```diff
@@ -0,0 +1,167 @@
+use std::str::FromStr;
```
Comment (radumarias) [inline src/crypto/cryptofs/test.rs]: call the module tests

**Review by radumarias — CHANGES_REQUESTED** [review]

File: src/crypto/cryptofs.rs (hunk 1 above)
Comment (someotherself) [reply src/crypto/cryptofs.rs]: These are fixed. Plus removed some dead fields in the state structs.
Will work on more docs and tests soon.

File: src/crypto/cryptofs/test.rs (hunk 3 above)
Comment (someotherself) [reply src/crypto/cryptofs/test.rs]: Module is now called tests.rs

**Pull request closed without merging**